tool/wiki/sync_runtime_parameter.py

```diff
@@ -0,0 +1,161 @@
+LINK_FILES     = [ "../../doc/wiki/Runtime-Parameters:-General.md",
+                   "../../doc/wiki/MPI-and-OpenMP.md",
+                   "../../doc/wiki/GPU.md",
+                   "../../doc/wiki/Runtime-Parameters:-Units.md",
+                   "../../doc/wiki/Initial-Conditions.md",
+                   "../../doc/wiki/Hydro.md",
+                   "../../doc/wiki/Gravity.md",
+                   "../../doc/wiki/Particles.md",
+                   "../../doc/wiki/Runtime-Parameters:-Cosmology.md",
+                   "../../doc/wiki/Chemistry-and-Radiation.md",
+                   "../../doc/wiki/Star-Formation.md",
+                   "../../doc/wiki/Feedback.md",
+                   "../../doc/wiki/Runtime-Parameters:-Timestep.md",
+                   "../../doc/wiki/Runtime-Parameters:-Refinement.md",
+                   "../../doc/wiki/Runtime-Parameters:-Interpolation.md",
+                   "../../doc/wiki/Outputs.md",
+                   "../../doc/wiki/Runtime-Parameters:-Miscellaneous.md"
+                 ]
+OUT_MD         = "Runtime-Parameters:-All.md"
+REPLACE_DICT   = { "NoMin_double":"None", "NoMax_double":"None", "NoDef_double":"None",
+                   "NoMin_int":"None", "NoMax_int":"None",
+                   "NoMin_long":"None", "NoMax_long":"None",
+                   "NoDef_str":"None", "Useless_str":"None",
+                   "Useless_bool":"None",
+                   "true":"1", "false":"0",
+                   "__DBL_MAX__":"1.79769313e+308", "Eps_double":"2.22507386e-308"
+                 }
+
+
+
+#====================================================================================================
+# Classes
+#====================================================================================================
+class parameter():
+    def __init__( self, string ):
+        name, description = self.get_name_description( string )
+        self.name         = name
+        self.link_name    = name
+        self.default      = []
+        self.minimum      = []
+        self.maximum      = []
+        self.description  = description
+        self.NAdd         = 0
+
+    def get_name_description( self, string ):
+        words       = string.split()
+        name        = words[0]
+        description = ""
+        for i in range(3, len(words)):
+            description += words[i] + ' '
+        return name, description[:-1]
+
+    def append_description( self, string ):
+        words = string.split()
+        self.description += "<br />" if self.name == "TESTPROB_ID" else ' '
+        self.description += ' '.join(words[1:])
+        return
+
+    def get_link_name( self, file_dict ):
+        for file_name in file_dict:
+            search_pattern = "[%s](#%s)"%(self.name, self.name)
+            if search_pattern in file_dict[file_name]:
+                self.link_name = "[[ %s \\| %s#%s ]]"%(self.name, file_name, self.name)
+                return True
+        return False
```

Comment:
This doesn't seem to work in my test. I add a new runtime parameter `TEST` to `Runtime-Parameters:-General.md` as
```
<a name="TEST"></a>
* #### `TEST` &ensp; (&#8805;0) &ensp; [0]
    * **Description:**
For testing purposes.
    * **Restriction:**
```

But I receive the warning message `Can not find description of TEST in all runtime parameter pages!`. I'll show you next week when we meet.

Reply:
I see. It works now. Thanks for the explanation!